src/main/java/net/minecraftforge/client/model/obj/OBJModel.java

```diff
@@ -1406,36 +1411,32 @@ else if (this.state instanceof TRSRTransformation)
             return quadList;
         }
 
-        private void put(ByteBuffer buffer, VertexFormatElement e, Float... fs)
-        {
-            Attributes.put(buffer, e, true, 0f, fs);
-        }
-
-        @SuppressWarnings("unchecked")
-        private final void putVertexData(UnpackedBakedQuad.Builder builder, Vertex v, TextureCoordinate t, Normal n, TextureAtlasSprite sprite)
+        private final void putVertexData(UnpackedBakedQuad.Builder builder, Vertex v, Normal faceNormal, TextureCoordinate defUV, TextureAtlasSprite sprite)
         {
             for (int e = 0; e < format.getElementCount(); e++)
             {
                 switch (format.getElement(e).getUsage())
                 {
                     case POSITION:
-                        builder.put(e, v.getPosition().x, v.getPosition().y, v.getPosition().z, v.getPosition().w);
+                        builder.put(e, v.getPos().x, v.getPos().y, v.getPos().z, v.getPos().w);
                         break;
                     case COLOR:
-                        float d = LightUtil.diffuseLight(n.normal.x, n.normal.y, n.normal.z);
+                        float d;
+                        if (v.hasNormal()) d = LightUtil.diffuseLight(v.getNormal().x, v.getNormal().y, v.getNormal().z);
```

Comment:
Pretty sure these sameline ifelse's go against forge style. Plus they are unreadable. No one is counting lines of code, use braces.